% Figure / table inputs

\chapter{Approach and Custom Software solutions} \label{chap:3}
\todo[inline]{replace CITEME tags in Approach and Custom Software solutions}


Frequently, \gls{functional-genomics} approaches contain problems or subproblems that require generating biologically meaningful gene-sets that pertain to the focus of the particular phenomenon being targeted.
Optimizing signal-to-noise ratios in data is a major challenge to generating useful gene-sets.
This project aims to generate meaningful gene-sets through integrating information from four supporting data-types:

\begin{itemize}
    \item comparative genomics ($N$-way 1-to-1 ortholog relationships)
    \item comparative transcriptomics (identical \gls{RNA-Seq} experiments performed using divergent mosquito species with the shared trait of \gls{hematophagy})
    \item phylogenetics (estimated divergence since last common ancestor)
    \item putative regulatory mechanisms (predicted \gls{TFBS} profiles)
\end{itemize}



\section{Preliminary data preparation} \label{sec:prelim-data}

Four foundational data-sets were compiled that represent Genomics, Transcriptomics, Phylogenetics, and Putative Transcriptional Regulatory information.

\paragraph*{Comparative Genomics:}

The initial gene-set for this project was limited to genes that have exactly one matching 1-to-1 ortholog in \Aa, \Ag, \As, and \Cq.
The purpose of this constraint is to maximize the probability of conserved function between genes analyzed. 

This preliminary gene set was obtained using the orthoDB7 \url{http://cegg.unige.ch/orthodb7} website by first selecting the node representing only the mosquito branch of the phylogeny \cite{Waterhouse2013}.
Then, the following requirements were enforced when retrieving ortholog relationship results 

\begin{quotation}
    \texttt{co-ortholog copy-number: AAEGY=1, AGAMB=1, ASTEP=1, CQUIN=1}
\end{quotation}

\input{figures/approach-chart}

The results of the query are provided as file \texttt{orthodb7\_mosqs\_20130918.txt} in the supplementary data files associated with this dissertation \cite{Dunn2013dissSupl}.
 
Custom python code was used to extract the 1-to-1 orthologs for \Aa, \Ag, and \Cq\ (Figure \ref{fig:approach-chart}).
The coded process is provided as an executable IPython notebook located at \url{http://goo.gl/EM1pHL} for in-depth inspection as well as for the purpose of openness and reproducibility.

Briefly, the results table was filtered for data pertaining to \Aa, \Ag, or \Cq.
Then I built a graph-representation of the data such that each orthoDB ortholog group could be queried to ensure that exactly one gene was represented for each species of interest.
These genes represent gene-set that formed the starting point for the rest of my approach.
The filtered set is available as supplementary file \texttt{AaAgCq\_1to1\_orthos.tsv} \cite{Dunn2013dissSupl}.

It should be noted that while this set contains only genes from the three species included in this document, each group should also have exactly one ortholog from \As\ as well which represents an additional constraint intended to limit the ortholog-sets to those most likely to fulfill the same function in each species.

\paragraph*{Comparative Transcriptomics:}

Whole RNA was extracted from dissected midguts of 16 female individuals of each species among the following time points: \gls{NBF} (0), 2, 4, 6, 8, 10 h \gls{PBM} (Protocol \ref{prot:RNA_Judy}) to identify and analyze orthologous genes that exhibit correlated \glspl{mAP} in the midguts of females directly following the ingestion of a bloodmeal.


\paragraph*{Phylogenetics:}

The phylogenetics aspect of this approach consists of using the estimated phylogenetic distances between each species to weight the comparisons between data correlations calculated for each pairwise 1-to-1 ortholog duo.
This is encoded as $d$ in the \gls{PTCI} expression defined in Equation \ref{eq:ptci} (see Section \ref{sec:ptci}).

The values of $d$ for each species-pair have been estimated previously (Table \ref{tab:evo-ranges}) (compiled in \citet{Sieglaff2009}).

\input{tables/evo-ranges}


\paragraph*{Putative Transcriptional Regulatory Control Mechanisms:}

Part of the \gls{PTCI} includes correlation of a \gls{TFBS} scanning signature between ortholog pairs.
If the \gls{PTCI} is calculated using a set of \gls{TFBS} that are known to interact with the \gls{20E} signaling cascade, then this portion of the \gls{PTCI} acts as a rough proxy for correlation with putative activity of the \gls{20E} cascade.

The \gls{TFBS} profiles are obtained through scanning the 2000 bp regions 5' to the annotated start of transcription for each gene using the \gls{MOODS} application wrapped in custom Python code \cite{Pizzi2009}.
The scoring method used was log-odds.
The final \gls{TFBS} profile score for a gene-motif combination consists of the sum of all site-scores with positive log-odds scores.
In this way the gene-motif score represents the count of all site-scores that fit the motif model better than would be expected after considering the background sequence-composition of the putative promoter-regions weighted by the log-odds value of each site-score.
For example, for motif model $m$, if one sequence had three sites that each received log-odds scores of 0.33, then the gene-motif score is approximately 1.
Alternatively, if a second sequence has a single site that matches $m$ with a positive score of 1, both sequences would have approximately equal gene-motif scores for $m$.
Just as the \gls{mAP} for a particular gene consists of 5 values that represent the \gls{FPKM} score for that gene at 5 time-points, the \gls{TFBS} profile of a gene consists of $N$ values (the number of motif models considered) that represent the sum of sites in the putative promoter region that received a positive log-odds value weighted by the log-odds value of each site.

This is an imperfect solution as \gls{ChIP-Seq} using antibodies to specific \glspl{TF} in each species is preferable\footnote{those that are presently associated with \gls{20E} for instance}.
However this was impractical for this work due to the absence of suitable antibodies and the cost of developing and executing all assays in each species.


\section{The phylogenetic transcriptional correspondence index (PTCI)} \label{sec:ptci}

The \gls{PTCI} (Equation \ref{eq:ptci}) attempts to provide a metric that combines multiple pairwise \glspl{Pcc} ($r$) relating to multiple comparison dimensions for each 1:1 ortholog-pair then weight it by the relative evolutionary divergence times of each pairwise comparison ($d$).
%
\input{equations/ptci}
%
In Equation \ref{eq:ptci}, the two comparison dimensions being combined are the similarity between an ortholog-pair with respect to \gls{mAP} ($r_{x}$) and \gls{TFBS} profile signature ($r_{t}$).
The pairwise species-divergence is represented by $d$.
However, the divergence times must be scaled using function $w()$ that prevents the \textbf{much} larger values of raw divergence values from dominating the index.
I have chosen to constrain the range of $w(d)$ between $1$ (no reward) and $1.1$ (a reward of 10\%).
This range was chosen to allow the evolutionary distance to affect the ranking of an ortholog-set without having a dominating effect.
I explored other constraint settings, but this range appeared to be a good compromise (not shown).%
%
\input{figures/ptci-space}%
%
The \gls{TFBS} data is derived through imperfect and non-empirical means\footnote{They are non-empirical in that they are statistical predictions rather than the results of physical experiments.}.
Therefore, they should not carry the same significance as the information encoded in the \glspl{mAP}.
I explored several methods for combining the two correlations.
Equation \ref{eq:ptci} values the \gls{TFBS} correlations at 50\% of the \gls{mAP} correlations.
This avoids scenarios in which an ortholog-pair that has $r_{x} = 0$ and $r_{t}=1$ ranks the same as if the values were opposite while still allowing the \gls{PTCI} to be positive overall\footnote{Positive \gls{PTCI} implies positive combined correspondence between the ortholog correlations which may still be valid even with the failure to detect correlation between \glspl{mAP} when combined with the detection of very strong correlation between \gls{TFBS} profiles.} but rank below ortholog-pairs with $r_{x} = 1$ and $r_{t}=0$ (Figure \ref{fig:ptci-space}).
Figure \ref{fig:ptci-space} provides detailed illustrations of the effects of weighting the \gls{TFBS} information at 50\% of the \gls{mAP} data across the spectrum of possible $r_{x}$ and $r_{t}$ value-combinations.

\input{figures/nway-ortholog-graph}

The \gls{PTCI} fundamentally is a pairwise index.
The $N$-way 1:1 ortholog-sets in aggregate are scored by calculating the arithmetic mean of the respective pairwise 1:1 ortholog-sets (Figure \ref{fig:nway-ortholog-graph-ptci}).
Figure \ref{fig:nway-ortholog-graph-model} illustrates how a hypothetical 3-way 1:1 ortholog-set is related by 1:1 orthology between three species.
This is the fundamental graph-relationship representation that is modeled in the \gls{gFunc} code\footnote{See Section \ref{sec:custom-software}.}.






\section{Custom Software Solutions}\label{sec:custom-software}

Two core bodies of software were developed for this project as well as numerous smaller custom Python scripts.
\gls{blacktie} is a python package that automates data flow between and record keeping for analysis runs of the components of the third-party analysis pipeline dubbed the \gls{TuxProt} \cite{Trapnell2012}.
The purpose of the second software project, the \gls{gFunc}, is to provide a framework with which to integrate many, disparate data types to facilitate the synthesis of functional genomics analyses that make use of as many types of data as may be useful.

\gls{blacktie}'s code and documentation have been made available to the public.
It is hosted as a \gls{Git} project with the complete history of code commits included and available for easy open source improvement and collaboration.
It is also available as an easily installable package from the \gls{PyPI} for general users.
It has been downloaded and installed over 2000 times as of this time with its latest version (v0.2.1.2) being downloaded at least 327 times (according to \gls{PyPI}).

\gls{gFunc} is based around an existing, well known, and well maintained network-graph framework called networkx \cite{Hagberg2008}.
\gls{gFunc} uses this code library as a database backbone and provides a framework for installing common genomic data-types into the graph structure as well as a convention for extending the core objects of \gls{gFunc} to handle new types of data.
The project is not formally released.
After using it for my own work, I have learned more about how best to organize the code.
Once it has been re-factored to incorporate the lessons learned here (mainly that the core objects should be simplified as much as possible and more of the data management should be coded directly with networkx rather than re-implementing certain functions), it will be released to the public using the same mechanisms as described for \gls{blacktie}.


\subsection{Blacktie RNA-Seq pipeline}
Leveraging multiple fastQ files of \gls{RNA-Seq} reads into a coherent picture of gene expression and transcript models is a multi-step process.
Various analysis methods and corresponding programs exist and continue to be developed (DESeq \cite{Anders2010}, edgeR \cite{Robinson2010a}, eXpress \cite{Roberts2013}, MISO \cite{Katz2010}, Oases \cite{Schulz2012}, Trans-ABySS \cite{Robertson2010}).
The \gls{TuxProt} is a popular method with good documentation and user-support \cite{Trapnell2012}.
It consists of a coordinated data-flow through multiple complementary but stand-alone programs: \gls{tophat}, \gls{bowtie}, \gls{cufflinks}, \gls{cuffmerge}, \gls{cuffdiff}, and finally \gls{cummeRbund} \cite{Kim2013}, \cite{Langmead2012}, \cite{Trapnell2010}, \cite{Goff2012}.
This requires the organization and coordination of many files of different types through many different program calls and output steps (Figure \ref{fig:tuxedo}).
Each step might take hours or days depending on the size of input data.
A further complication inherent to the use of swiftly evolving academic software is that new versions of the analysis software may be released after the analyses have been run but before the work has been published.
It is not uncommon for the latest version to give appreciably different results than the version originally used.

\input{figures/tuxedo}

The purpose of \gls{blacktie} is to streamline and simplify the complex task of analyzing full \gls{RNA-Seq} experiments using these programs; to record automatically settings used and program output messages so that users may track them later to data; provide a base-set of functions and classes that will allow users to create custom pipelines easily by editing a single file (or if they want: writing their own custom scripts).
For example, \gls{blacktie} makes the process of re-running an analysis from the point of an updated program more reliable because all settings and source-files are recorded automatically and made available alongside the original output.


Some other features include:

\begin{enumerate}
    \item simple installation
    \item simple command line interface that allows one to fully automate and reliably repeat their analysis of \gls{RNA-Seq} data using the \gls{TuxProt}
    \item ability to send email updates to the user
    \item intelligently continue with as much of the analysis as possible if a single branch of the pipeline fails
    \item analyze multiple, complex \gls{tophat}/\gls{cufflinks} experiments at once using a single command
    \item automatically generates \gls{SGE} scripts for use with a computing cluster
    \item checks the user's system for an R installation (required to run \gls{cummeRbund})
    \item checks for \gls{cummeRbund} library and walks user through installation if it is not currently installed
    \item automatic preliminary  \gls{cummeRbund} Quality Control, Basic Differential Expression, and Basic Pattern Discovery plots using cummeRbund
\end{enumerate}



\gls{blacktie} aims to bring automated, reproducible \gls{RNA-Seq} analysis with built-in record keeping to more researchers so that quality results may be obtained faster, easier and will contain fewer mistakes.



\subsection{gFunc framework for the integration and analysis of disparate genomic-scale data-types}

A common problem in the era of big data is how to incorporate many types of genome-scale data so that that the question at hand is meaningfully informed.
Relational databases such as MySQL or PostgreSQL have been used as data storage and analysis workhorses for decades at least.
Relational databases and graph-style databases each have performance advantages and disadvantages depending on the structure of the data and the most common operations expected to be carried out on the data.
However, a major advantage of graph-style models lies in the similarity that is shared with how many biologists conceptualize data and interactions between data entities.
A gene or protein can easily be conceptualized as a node with edges representing interactions or relationships\footnote{Edges may easily be conceptualized as many biologically relevant relationships such as homology, physical interactions, ancestor} between the nodes (Figure \ref{fig:ecrusp-graph}) \cite{Franceschini2013}.
\input{figures/ecrusp-graph}


\subsubsection{Package Organization}
The code is organized around three fundamental types of classes.
The first type is made up of the data classes.
These model how data pertaining to nodes or edges are stored and accessed.
The second type of class consists of parser classes.
These consume a source file and convert the information into forms that can be loaded into data classes.
The third type are the graph tool classes (GraphBuilder, and GraphHandler).
These deal with mapping the parsed data onto the graph structure and help access the information from that graph as analyses are being performed, respectively.


The flexibility of this organization lies in the graph construction process.
The data parser classes encode specific information on how a single data-type is digested.
They all inherit a set of basic methods that control how data should be mapped to the graph, whether it is edge-data or node-data.
Because they all have this same graph-facing interface, the GraphBuilder does not need to know what kind of data the parser represents.
It simply is provided with a parser class, and it consumes that parser as the information streamed through the parser is correctly added to the graph.
Because of this format, it is straightforward to adapt the \gls{gFunc} system to accept new data-types.
One simply needs to write a parser class that inherits from the base parser class and include the correct rules for parsing the new data-type.
That new parser can be included in a list of existing parser classes to be provided to the GraphBuilder and the new data will be correctly added into the graph structure.

The GraphHandler stores the graph and provides conveniences for accessing its contents, such as node and edge dictionaries.


\section{gFunc-based analysis}
Once the preliminary data preparation is accomplished (Section \label{sec:prelim-data}), the data is loaded into a \gls{gFunc} graph through a script (\texttt{gfunc\_build\_n\_way\_one2one.py} \cite{Dunn2013dissSupl}) that coordinates the initiation of relevant parsers and provides them to a GraphBuilder object that maps the data onto the 3-way 1:1 ortholog-relationships.
\glspl{FDR} are empirically estimated by calculating all 3-way-\gls{PTCI} values first using the real 3-way 1:1 ortholog relationships, then repeating the analysis many times with randomized (null) 3-way 1:1 orthology relationships.
\alert{(for what follows: should i refer to the figure from the next chapter here?  Its kinda like having results in the methods section?)}
The \gls{FDR} is then defined, for a given 3-way-\gls{PTCI} threshold, as the median value of the null 3-way \glspl{PTCI} divided by the actual 3-way \gls{PTCI} (\texttt{PTCI\_testing\_rsrd\_1.0\_1.1\_new\_data\_Jaspar\_nr\_insect\_20130918\_orthodb7.ipynb} \cite{Dunn2013dissSupl}).
This information is used to define an acceptable 3-way-\gls{PTCI} threshold that balances selectivity with sensitivity.
Given a 3-way-\gls{PTCI} threshold, a filter script (\texttt{Getting\_and\_using\_filtered\_gfunc\_genes\_ecr\_team\_OR\_jaspar\_insect\_20120918\_orthodb7.ipynb} \cite{Dunn2013dissSupl}) is applied to extract the 3-way 1:1 ortholog sets that are greater than or equal to the threshold.

This process is carried out using two sets of \gls{TFBS} profile data: one that covers the entire JASPAR \gls{TFBS}-model database for insects (\texttt{JASPAR\_CORE\_2009\_insects.meme}  \cite{Dunn2013dissSupl}), another that is limited to 12 \gls{TFBS}-models that are thought to participate in \gls{20E}-regulated transcriptional regulatory mechanisms (\texttt{EcR\_team.meme} \cite{Dunn2013dissSupl}).
The later is meant to provide a set of 3-way 1:1 orthologs generated with a focus toward a particular regulatory program while the former is intended to capture a more general set.
The union of the two result-sets are taken forward to the Characterization Stage.



\section{Characterization of results} \label{chap:3-sec:characterization-of-results}
The \gls{gFunc} analysis yields a set of the top midgut transcriptionally corresponding 3-way 1:1 ortholog-sets.
Two types of characterization were carried out on these ortholog-sets.
The characterization method used \gls{Argot2} to obtain functional annotations of the ortholog-sets by comparing the protein sequences of the 3-way 1:1 ortholog-sets against previuosly annotated proteins (\gls{Swiss-Prot}) and hidden markov-models of described protein domains (\gls{Pfam}) \cite{Boeckmann},\cite{Punta2012},\cite{Falda2012}.
The second method consisted of grouping the \glspl{mAP} by similarity using k-means clustering.

\subsection{Functional inference of filtered 3-way 1:1 ortholog sets}

\gls{Argot2} was used to infer functional annotations for the selected proteins of the three mosquito species.
First, the peptide sequences must be compared with well characterized databases of functinally annotated data.
\gls{Argot2} suppports the use of either the Swiss-Prot\footnote{Swiss-Prot is the manually annotated and reviewed section of the UniProt Knowledgebase.} database of protein sequences or the \gls{Pfam}\footnote{The Pfam database is a large collection of protein families, each represented by multiple sequence alignments and \glspl{HMM}.} database of protein domain models or both.
Here both were used.

The following blastp settings were used against the \gls{Swiss-Prot} proteins, as requested by the \gls{Argot2} instructions:

\begin{Verbatim}
blastp -outfmt "6 qseqid sseqid evalue " -query your_sequences \
      -db Uniprot -out output_file
\end{Verbatim}

Likewise, the following HMMER3 settings were used against the \gls{Pfam} domain models\footnote{The Pfam-A (hand-currated models) and Pfam-B (automated curration) databases were combined as suggested by \gls{Argot2}.}, as requested by the \gls{Argot2} instructions:


\begin{Verbatim}
hmmscan --tblout output_file P-fam_database your_protein_sequences
\end{Verbatim}

Both sets of results were then compressed into ``zip'' files and uploaded to the \gls{Argot2} servers for analysis.

\subsection{Clustering of filtered 3-way 1:1 ortholog sets} \label{chap:3-sec:clustering-of-filtered-ortholog-sets}

Clusters were generated by custom python code that used Biopython's k-means clustering (version 1.62) \cite{Cock2009}.
\gls{mAP} data was log transformed after 1 FPKM was added to all data to remove zeros ($log_{10}(\mathrm{FPKM}+1)$). 
K-means was then applied using the arithmetic mean as the center for cluster definition.
Twenty-three clusters ($k = 23$) were used.
The clustering was performed using the \glspl{mAP} from \Ag\ because this species is approximately equidistant from the other two with regard to a last common ancestor.
Four of the resulting clusters were chosen for further description in Chapter \ref{chap:4} and the orthologs of the corresponding members of each cluster from \Aa\ and \Cq\ were retrieved and combined with those from \Ag.
The four clusters were chosen because they exhibit a major change between \gls{NBF} and 4h \gls{PBM}\footnote{There is a small but reproducible peak in \gls{20E} titers that occurs 4h \gls{PBM} \cite{Hagedorn1975} and coincides with a peak in posterior midgut aminopeptidase activity that has been noted in \As\ \cite{Billingsley1991}; an initiation of the increase of late trypsin in the midgut \cite{Barillasmury1991},\cite{Graf1988}; and an increase of JH-esterase activity that causes a rapid decline of \gls{JH} \cite{Shapiro1986}.} samples that may indicate the \glspl{mAP} contained in these clusters are influenced by \gls{20E}-regulated transcriptional regulatory mechanisms.
The four patterns can be described roughly as ``up-regulated \textit{at} 4h \gls{PBM}'', ``down-regulated \textit{at} 4h \gls{PBM}'', ``up-regulated \textit{after} 4h \gls{PBM}'', and ``down-regulated \textit{after} 4h \gls{PBM}''.




%%% Local Variables: ***
%%% mode: latex ***
%%% TeX-master: "thesis.tex" ***
%%% End: ***
